pkgs/development/python-modules/xdot/default.nix

```
@@ -10,29 +10,38 @@
   graphviz,
   gtk3,
   numpy,
+  packaging,
+  python3Packages,
```

Since you `callPackage` this file from the `python-packages.nix` file, you should be able to use `setuptools` directly, as that is more common to do and I suspect a bit more correct.

```suggestion
  setuptools,
```

---

Definitely more correct. Otherwise you get the Python 3.12 version of setuptools even if you build `python311Packages.xdot`.

---

> Definitely more correct. Otherwise you get the Python 3.12 version of setuptools even if you build `python311Packages.xdot`.

Really? I thought that `callPackage` as defined in `python-packages.nix` passes `python3Packages = self` or something alike.

---

No, the only thing it passes is `python`.